compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -1131,22 +1160,19 @@ LogicalResult aie2xclbin(
     const std::string &amdAIEInstallDir,
     const std::optional<std::string> &InputXCLBin,
     const std::optional<std::string> &ukernel) {
-  PassManager pm(ctx, mlir::ModuleOp::getOperationName());
+  PassManager pm(ctx, AIE::DeviceOp::getOperationName());
   applyConfigToPassManager(pm, printIRBeforeAll, printIRAfterAll,
                            printIRModuleScope, timing);
-  // generateNPUInstructions
-  pm.addNestedPass<AIE::DeviceOp>(
-      mlir::iree_compiler::AMDAIE::createAMDAIEDmaToNpuPass());
-  if (failed(pm.run(moduleOp)))
-    return moduleOp.emitOpError(": NPU Instruction pipeline failed");
-
-  std::optional<ArrayRef<uint32_t>> npuInstructions =
-      cast<DenseUI32ResourceElementsAttr>(
-          (*moduleOp.getOps<xilinx::AIE::DeviceOp>().begin())
-              ->getAttr("npu_instructions"))
-          .tryGetAsArrayRef();
-  if (!npuInstructions)
-    return moduleOp.emitOpError(": No NPU instructions in device op");
+  if (failed(pm.run(deviceOp)))
+    return deviceOp.emitOpError(": NPU Instruction pipeline failed");
```

Comment:
style guide nit (and I know it was here before): conditionals that don't fit on one line get braces (other @jtuyls will hunt you down)